### ISSUE TITLE
[Haskell] Add Track Structure files: CLI and Debug

### ISSUE THREAD

**Issue by sshine** [exercism/v3#544]
## Description

Before commencing with this task,

- [x] Clarify the relationship between [v2 `docs/*.md`](https://github.com/exercism/haskell/tree/master/docs) and [v3 `shared/.docs/*.md`](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#track-structure).

As described in [`docs/concept-exercises.md#track-structure`](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#track-structure),

- [x] Create `languages/haskell/exercises/shared/.docs/cli.md` (see: [C# example](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/shared/.docs/cli.md))
- [x] Create `languages/haskell/exercises/shared/.docs/debug.md` (see: [C# example](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/shared/.docs/debug.md))

## Resources to refer to

The Haskell v2 track currently lists the following related resources on the website:

- [Installing Haskell](https://exercism.io/tracks/haskell/installation)
- [Running the Tests](https://exercism.io/tracks/haskell/tests)
- [Learning Haskell](https://exercism.io/tracks/haskell/learning)
- [Useful Haskell Resources](https://exercism.io/tracks/haskell/resources)
- [Remind me how to set up the CLI](https://exercism.io/cli-walkthrough) (general, not Haskell-specific)

The markdown for these files can be found [here](https://github.com/exercism/haskell/tree/master/docs).

Additionally, some of these resources may be worth something:

- https://docs.haskellstack.org/
- https://typeclasses.com/phrasebook

### Contributing

Create a pull request for either one or both of the files above.

## Help

If you have any questions while adding the reference document, please post the questions as comments in this issue.

**Comment by sshine** [comment]
So v2's `docs/*.md` are the help links in the menu bar, and v3's `exercises/shared/.docs/{cli,debug}.md` are files that get appended to all problem descriptions. If any more clarification is needed, questions and answers can be provided here.

**Comment by zuzia-kru** [comment]
Hi there, I'd like to help with this issue. 
I'm not sure about one thing though - should those docs include all the information that existing resources include? I can't see things like installing or learning in the C# example, it only explains how to run included test suite. Can you help me understand better the scope?

Thanks

**Comment by sshine** [comment]
Hi @zuzia-kru, and sorry for the delayed response.

That is a good question that I don't know the answer to.

@ErikSchierboom, could you advise us?

**Comment by ErikSchierboom** [comment]
> I can't see things like installing or learning in the C# example, it only explains how to run included test suite.

Nope, those are not to be included at the moment. The purpose of the `debug.md` and `cli.md` documents are to be merged into a final `README.md` when the exercise is downloaded. This does not include the generic learning documents.

**Comment by zuzia-kru** [comment]
Thanks Erik and Simon! I'll work on a PR regarding this issue then :)

**Comment by aimorris** [comment]
@zuzia-kru Hello! Are you still working on a PR for this issue?

**Comment by zuzia-kru** [comment]
Hello @aimorris , yes I am, is that okay?

**Comment by aimorris** [comment]
@zuzia-kru That's great. Keep up the good work :)

**Comment by sshine** [comment]
Closing this issue as all items have been addressed in this issue and in #1235.

Thanks, @zuzia-kru!

This issue was my point of reference for the progress of Haskell track v3, so closing it means that current visible work is only a bunch of draft PRs I've made that I haven't gotten back to in a while.

If you're interested in doing more Haskell v3 stuff, @zuzia-kru, it'd probably relate to either "extracting concepts" or simply "defining some concept exercises." As this spells "create a curriculum" in my head, and I have somewhat high personal standards when it comes to learning material, this has kept me from progressing at all.
